apps/api/migrations/subscribers/remove-duplicated-subscribers/remove-duplicated-subscribers.migration.ts

```diff
@@ -0,0 +1,150 @@
+    const mergedSubscriber = mergeSubscribers(sortedSubscribers);
+    const subscribersToRemove = sortedSubscribers.filter((subscriber) => subscriber._id !== mergedSubscriber._id);
+
+    // eslint-disable-next-line no-console
+    console.log(
+      'Merged subscriber:',
+      mergedSubscriber._id.toString(),
+      'subscriberId:',
+      subscriberId,
+      'environmentId:',
+      environmentId.toString()
+    );
+
+    try {
+      await subscriberRepository.update(
+        {
+          _id: mergedSubscriber._id,
+          subscriberId: subscriberId,
+          _environmentId: environmentId,
+        },
+        {
+          $set: mergedSubscriber,
+        }
+      );
+
+      // eslint-disable-next-line no-console
+      console.log(
+        'Remaining subscriber updated with merged data for subscriberId:',
+        subscriberId,
+        'subscriberId:',
+        mergedSubscriber._id.toString(),
+        'environmentId:',
+        environmentId.toString()
+      );
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error('Error updating remaining subscribers:', err);
+    }
+
+    try {
+      // Delete all duplicates except the merged one
+      await subscriberRepository.deleteMany({
+        _id: { $in: subscribersToRemove.map((subscriber) => subscriber._id) },
+        subscriberId: subscriberId,
+        _environmentId: environmentId,
+      });
+      // eslint-disable-next-line no-console
+      console.log(
+        'Duplicates deleted for subscriberId:',
+        subscriberId,
+        'environmentId:',
+        environmentId.toString(),
+        'ids:',
+        subscribersToRemove.map((subscriber) => subscriber._id).join()
+      );
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error('Error deleting duplicates:', err);
+    }
+  }
+
+  // eslint-disable-next-line no-console
+  console.log('end migration - remove duplicated subscribers');
+
+  app.close();
+}
+
+// Function to merge subscriber information
+function mergeSubscribers(subscribers) {
+  const mergedSubscriber = { ...subscribers[0] }; // Start with the first subscriber
+
+  // Merge information from other subscribers
+  for (const subscriber of subscribers) {
+    const currentSubscriber = subscriber;
+    for (const key in currentSubscriber) {
+      // Skip internal and irrelevant fields
+      if (
+        [
+          '_id',
+          '_organizationId',
+          '_environmentId',
+          'deleted',
+          'createdAt',
+          'updatedAt',
+          '__v',
+          'isOnline',
+          'lastOnlineAt',
+        ].includes(key)
+      ) {
+        continue;
+      }
+
+      // Update with non-null/undefined values from subsequent subscribers
+      if (currentSubscriber[key] !== null && currentSubscriber[key] !== undefined) {
+        mergedSubscriber[key] = currentSubscriber[key];
+      }
```

Comment:
⚠ **issue:** I think that this part might be problematic for the `channels` field, which is an array, so the latter will override the first. Should we have here more custom logic and compare channels by `integrationId` and merge all the unique `deviceTokens` together? wdyt?

Reply:
Here, I wasn't sure how specific we wanted to be because we might say the same with 'data.' but, since we're uncertain about the type of data being stored, we could check by object size. Yet again, I wasn't certain if this level of detail was what we were aiming for.